app/assets/src/components/ui/containers/DataTooltip.jsx

```diff
@@ -25,8 +25,10 @@ const DataTooltip = ({ data, subtitle, title, singleColumn }) => {
                 <React.Fragment>
                   <div className={cs.dataTooltipLabel}>{keyValuePair[0]}</div>
                   <div className={cs.dataTooltipValue}>
-                    {/*Use .name if value is an object (e.g. location object)*/}
-                    {isObject(keyValuePair[1])
+                    {/*Use .name if value is an object (e.g. location object) AND it's
+                      an object with a name property */}
+                    {isObject(keyValuePair[1]) &&
+                    keyValuePair.hasOwnProperty("name")
```

Comment:
I'm assuming that you have an object here which doesn't have a name property. Wouldn't directly displaying the object render something like `[Object object]`? I'm a little confused what's going on here. Maybe you should hide the key value pair if it's an object without a name property?

I don't like this convention of passing in objects and then inferring the name. (I know this wasn't you). A better approach is to explicitly pass the string that you want displayed into `<DataTooltip>`.

For example, instead of passing `data={["Key", { "name": "Value", "foo": "bar" }]}`, just pass `data={["Key", "Value"]}`

Reply:
The "object" that I pass in `AMRHeatmapVis` is either just a plain string or a short piece of React. `keyValuePair` is an array of what is displayed in the tooltip, e.g. in this image:

![Screen Shot 2019-08-14 at 5 55 20 PM](https://user-images.githubusercontent.com/24234461/63065916-c46b3480-bebc-11e9-897b-bcb211759987.png)

The names on the left are the first entry in the array, and the values on the right are the second entry in the array. The code to generate the bottom section in `AMRHeatmapView` for example is 

```
return [
    metric.text,
    metric.value === selectedOptions.metric ? <b>{value}</b> : value,
];
```

So it sometimes is an object, that object is sometimes a bit of React, and sometimes that object is something else (like with the phylo tree I think?). Since multiple files (I count at least 5) use `DataTooltip` changing the approach would require quite a bit of refactoring. So for an easier to read fix (and I agree it should be rewritten to be easier to read and handle) that would take some time. I don't have much, so perhaps someone else could implement a longer term fix.

Reply:
Okay, I understand. I forgot that you can also pass React elements, which are interpreted as objects. Thanks for clarifying!

Could you change the comment to something like: `Use .name if value is an object (e.g. location object) with a name property. Display React elements normally.`? So it's easier to make the connection.